sound/hda/hdac_controller.c

```diff
@@ -555,8 +555,8 @@ int snd_hdac_bus_handle_stream_irq(struct hdac_bus *bus, unsigned int status,
 			sd_status = snd_hdac_stream_readb(azx_dev, SD_STS);
 			snd_hdac_stream_writeb(azx_dev, SD_STS, SD_INT_MASK);
 			handled |= 1 << azx_dev->index;
-			if (!azx_dev->substream || !azx_dev->running ||
-			    !(sd_status & SD_INT_COMPLETE))
+			if ((!azx_dev->substream && !azx_dev->cstream) ||
+			    !azx_dev->running || !(sd_status & SD_INT_COMPLETE))
```

Comment:
again this won't work if we disable SOF_HDA

Reply:
This code is SOF-agnostic. Why should this piece care about said config?

Reply:
sound/hdac has a GPLv2 license.
Without SOF_HDA we are able to keep SOF working with dual-license code.
If you stick the probe work in a GPLv2 licensed library, you pretty much take the entire code with you as only working in GPLv2.